docs/ndcubesequence.rst

```diff
@@ -408,41 +391,167 @@ dimensions.shape ``(<Quantity 3.0 pix>, <Quantity 3.0 pix>, <Quantity
+
+This will create a new class, ``MySequence``, which contains all the functionality of
+`~ndcube.NDCubeSequence` plus the plot methods you've defined in ``MySequencePlotMixin``.
+
+Extracting and Plotting NDCubeSequence Data with Matplotlib
+-----------------------------------------------------------
+
+In order to produce plots (or perform other analysis) outside of the ``ndcube`` framework,
+it may be useful to extract the data from the `~ndcube.NDCubeSequence` into single
+`~numpy.ndarray` instances.
 In the above examples we defined the `my_sequence` `~ndcube.NDCubeSequence` object.::
 
     >>> # Print dimensions of my_sequence as a reminder
     >>> print(my_sequence.dimensions)
     (<Quantity 3. pix>, <Quantity 3. pix>, <Quantity 4. pix>, <Quantity 5. pix>)
 
-In this section we will use this object to demonstrate extracting data arrays from `~ndcube.NDCubeSequence` objects.
-For example, say we wanted to make a 4D array out of the data arrays within the `~ndcube.NDCubes` of `my_sequence`.::
+To make a 4D array out of the data arrays within the `~ndcube.NDCubes` of `my_sequence`.::
 
-    >>> # Make a single 4D array of data in sequence.
-    >>> data = np.stack([cube.data for cube in my_sequence.data])
+    >>> # Make a single 4D array of data in sequence with the sequence axis as the 0th.
+    >>> data4d = np.stack([cube.data for cube in my_sequence.data], axis=0)
     >>> print(data.shape)
     (3, 3, 4, 5)
 
-If instead, we want to define a 3D array where every `~ndcube.NDCube` in the `~ndcube.NDCubeSequence` is appended
-together, we can use `numpy`'s `vstack` function::
+The same applies to other array-like data in the `~ndcube.NDCubeSequence`, like 
+``uncertainty`` and ``mask``.
+If instead, we want to define a 3D array where every `~ndcube.NDCube` in the
+`~ndcube.NDCubeSequence` is appended along the ``common_axis``, 
+we can use `numpy.concatenate` function::
 
     >>> # Make a 3D array
-    >>> data = np.vstack([cube.data for cube in my_sequence.data])
+    >>> data3d = np.concatenate([cube.data for cube in my_sequence.data],
+                              axis=my_sequence._common_axis)
     >>> print(data.shape)
     (9, 4, 5)
 
-Finally, we can also create 3D arrays by slicing `~ndcube.NDCubeSequence` objects.
-Here we slice the `~ndcube.NDCubeSequence` along the fastest-changing dimension::
-
-    >>> # Slice sequence to make 3D array
-    >>> data = np.stack([cube[2].data for cube in my_sequence.data])
-    >>> print(data.shape)
-    (3, 4, 5)
+Having extracted the data, we can now use matplotlib to visualize it.
+Let's say we want to produce a timeseries of how intensity changes in a
+given pixel at a given wavelength.  We stored time in ``my_sequence.global_coords``
+and associated it with the ``common_axis``.  Therefore, we could do::
+
+    >>> import matplotlib.pyplot as plt
+    >>> # Get intensity at pixel 0, 0, 0 in each cube.
+    >>> intensity = np.array([cube.data[0, 0, 0] for cube in my_sequence])
```

Comment:
can't we use the stack here rather than the list comprehension?

Reply:
I think we still need list comprehension to extract the data arrays one by one from the cubes in the sequence.  And in this case, since we are extracting scalar values, we don't need stack on top of the list comprehension.